apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/DrawerFooter.tsx

```diff
@@ -28,17 +29,64 @@ export function DrawerFooter({ provider }: { provider: string }) {
     getProvideData();
   }, [provider]);
 
+  const ptxSwapLiveAppKycWarning = useFeature("ptxSwapLiveAppKycWarning");
   const url = providerData?.termsOfUseUrl;
-  const onLinkClick = useCallback(() => openURL(url!), [url]);
+  const onLinkClick = useCallback((link: string) => openURL(link!), []);
   if (!url) {
     return null;
   }
   const providerName = getProviderName(provider);
-  const acceptTerms =
-    providerName === "Exodus"
-      ? "DeviceAction.swap.exodusAcceptTerms"
-      : "DeviceAction.swap.acceptTerms";
-
+  let acceptTerms: string;
+  let linksComponents: React.JSX.Element[] = [];
```

Comment:
this whole thing including the if and switch cases to me looks like something that should be memorized 🤔  and since the component is always the same as far as I can see, I'd be in favour of only returning the props from here and render the component as before 🤔